src/Middleware/HttpLogging/src/FileLoggerProcessor.cs

```diff
@@ -0,0 +1,269 @@
+            while (!_cancellationTokenSource.IsCancellationRequested)
+            {
+                while (_messageQueue.TryTake(out var message))
+                {
+                    _currentBatch.Add(message);
+                }
+                if (_currentBatch.Count > 0)
+                {
+                    try
+                    {
+                        await WriteMessagesAsync(_currentBatch, _cancellationTokenSource);
+                    }
+                    catch (Exception ex)
+                    {
+                        Log.WriteMessagesFailed(_logger, ex);
+                    }
+
+                    _currentBatch.Clear();
+                }
+                else
+                {
+                    try
+                    {
+                        await Task.Delay(_flushInterval, _cancellationTokenSource.Token);
+                    }
+                    catch
+                    {
+                        // Exit if task was canceled
+                        return;
+                    }
+                }
+            }
+        }
+
+        private async Task WriteMessagesAsync(List<string> messages, CancellationTokenSource cancellationTokenSource)
+        {
+            // Files are grouped by day, and written up to _maxFileSize before rolling to a new file
+            DateTime today = DateTime.Now;
+            var fullName = GetFullName(today);
+            var fileInfo = new FileInfo(fullName);
+            if (!TryCreateDirectory())
+            {
+                // return early if we fail to create the directory
+                return;
+            }
+            var streamWriter = GetStreamWriter(fullName);
+
+            try
+            {
+                foreach (var message in messages)
+                {
+                    if (cancellationTokenSource.IsCancellationRequested)
+                    {
+                        return;
+                    }
+                    fileInfo.Refresh();
+                    // Roll to new file if _maxFileSize is reached
+                    // _maxFileSize could be less than the length of the file header - in that case we still write the first log message before rolling
+                    if (fileInfo.Exists && fileInfo.Length > _maxFileSize)
+                    {
+                        _fileNumber++;
+                        fullName = GetFullName(today);
+                        fileInfo = new FileInfo(fullName);
+                        streamWriter.Dispose();
+                        if (!TryCreateDirectory())
+                        {
+                            // return early if we fail to create the directory
+                            return;
+                        }
+                        streamWriter = GetStreamWriter(fullName);
+                    }
+                    if (!fileInfo.Exists || fileInfo.Length == 0)
+                    {
+                        await OnFirstWrite(streamWriter, _cancellationTokenSource);
```

Comment:
We should be consistently passing a `cancellationToken` arg through the methods and not referencing `_cancellationTokenSource` except for in Dispose().